addons/refuel/functions/fnc_refuel.sqf

```diff
@@ -29,6 +29,10 @@ _maxFuel = getNumber (configFile >> "CfgVehicles" >> (typeOf _target) >> QGVAR(f
     params ["_args", "_pfID"];
     _args params ["_source", "_sink", "_unit", "_nozzle", "_rate", "_startFuel", "_maxFuel", "_connectFromPoint", "_connectToPoint"];
 
+    if (_nozzle getVariable [QGVAR(isConnected), false]) exitWith {
```

Comment:
Sorry, your code doesn't reflect what you are saying. This PFH has to run while the `_nozzle` is connected.

`QGVAR(isConnected)` is set to `true` for `_nozzle` in [fnc_connectNozzleAction](https://github.com/acemod/ACE3/blob/master/addons/refuel/functions/fnc_connectNozzleAction.sqf#L94) which is calling `fnc_refuel`. So you would add the PFH and remove it next time it executes.

So either change it to `_source` and the PFH gets removed upon returning.
Or query `QGVAR(sink)` on `_nozzle` which is also set [here](https://github.com/acemod/ACE3/blob/master/addons/refuel/functions/fnc_connectNozzleAction.sqf#L93) but gets cleared on [disconnecting](https://github.com/acemod/ACE3/blob/master/addons/refuel/functions/fnc_disconnect.sqf#L26)